### PR TITLE
cat actions

### PR DESCRIPTION
**Pull request by andrewdoro**

### CODE REVIEW

File: components/CatModal/CatActions.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  const DeleteCat = async () => {
+    await deleteCat(zoneId, catId);
+    mutate(`/api/interest-zones/`);
+    mutate(`/api/interest-zones/${zoneId}`);
+  };
+
+  const UpdateCat = async (form: { values: FormValues }) => {
+    const body = {
+      gender: form.values.gender,
```
Comment (florea-razvan) [inline components/CatModal/CatActions.tsx:42]: please extract the code responsible for creating this body in a separate private function and call it for add / update / sterilize

File: components/CatModal/CatModalView.tsx [hunk 2]
```diff
@@ -33,134 +36,163 @@ const useStyles = createStyles((theme) => ({
...
+      ...(cat || {
+        gender: Gender.UNKNOWN,
+        observations:
+          'Enter here any observations, notes, that might help identify the cat more accurately.',
+        mediaLinks: [],
+      }),
+      volunteerName: (cat as SterilizedCat)?.volunteerName || '',
+      hospitalizationDate: new Date((cat as SterilizedCat)?.hospitalizationDate || Date.now()),
+      releaseDate: new Date((cat as SterilizedCat)?.releaseDate || Date.now()),
+      sterilizedStatus: (cat as SterilizedCat)?.releaseDate ? 'sterilized' : 'unsterilized',
```
Comment (florea-razvan) [inline components/CatModal/CatModalView.tsx:73]: do not forget to use here the values of the enum defined at the first comment

File: components/InterestZoneView/SterilizedCat.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+import useCatActions from '../CatModal/CatActions';
+
+interface Props {
+  cat?: SterilizedCat;
+}
+
+const SterilizedCat = ({ cat }: Props) => {
+  const { setModal, modal } = useContext(ModalContext);
+  const { setCat } = useContext(CatContext);
+  const [AddCat, UpdateCat, DeleteCat, SterilizeCat] = useCatActions(cat?._id!);
```
Comment (florea-razvan) [inline components/InterestZoneView/SterilizedCat.tsx:41]: is SterilizeCat action needed?

File: styles/theme.ts [hunk 4]
```diff
@@ -15,18 +15,6 @@ const theme: MantineThemeOverride = {
       '#8C7821',
       '#594D15',
     ],
-    green: [
```
Comment (florea-razvan) [inline styles/theme.ts:18]: I guess these were not supposed to be deleted

**Review by florea-razvan — CHANGES_REQUESTED** [review]

File: components/InterestZoneModal/SterilizedCat.tsx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+            <Text component='span' weight={600} color='green'>
+              NOTE:{' '}
+            </Text>
+            {cat?.observations}
+          </Text>
+          <Group spacing='md' mt='xs'>
+            <Group spacing='xs'>
+              <ArrowBarRight />
+              <Text size='sm'>
+                {new Date(cat?.hospitalizationDate || ' ').toLocaleDateString('en-GB')}
```
Comment (florea-razvan) [inline components/InterestZoneModal/SterilizedCat.tsx:120]: use new Date() as fallback

**Review by florea-razvan — CHANGES_REQUESTED** [review]

**Review by florea-razvan — APPROVED** [review]

File: components/CatModal/useCatActions.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import { InterestZoneProviderContext } from '../Providers/ZoneProvider';
+import { FormValues } from './CatModalView';
+
+const useCatActions = (catId: string) => {
+  const { interestZone } = useContext(InterestZoneProviderContext);
+  const { mutate } = useSWRConfig();
+  const zoneId = interestZone?._id!;
+
+  function getBody(values: FormValues) {
+    let body: any = {
```
Comment (vasileorza) [inline components/CatModal/useCatActions.tsx:13]: Please look into adding typing to both the `function` and the `body` constant

File: components/HeaderGoogle/HeaderGoogle.tsx [hunk 7]
```diff
@@ -1,92 +1,106 @@
...
 
-const HeaderGoogle = (props: Props) => {
+const HeaderGoogle = ({ searchPosition }: Props) => {
   const { user } = useUser();
-  const [addressName, setAddressName] = useState<string>("");
+  const [addressName, setAddressName] = useState<string>('');
 
   const searchPlace = (): void => {
-    props.searchPosition(addressName);
+    console.log(addressName);
```
Comment (vasileorza) [inline components/HeaderGoogle/HeaderGoogle.tsx:27]: Please remove redundant `log`

File: components/CatModal/CatModalView.tsx [hunk 8]
```diff
@@ -33,134 +36,162 @@ const useStyles = createStyles((theme) => ({
...
-    releaseDate: (cat as SterilizedCat)?.releaseDate || '',
+  const theme = useMantineTheme();
+  const form = useForm<FormValues>({
+    initialValues: {
+      ...(cat || {
+        gender: Gender.UNKNOWN,
+        observations:
+          'Enter here any observations, notes, that might help identify the cat more accurately.',
+        mediaLinks: [],
+      }),
```
Comment (vasileorza) [inline components/CatModal/CatModalView.tsx:68]: Not sure if it is an improvement, but I'd write the code as follows
`
initialValues: {
        gender: Gender.UNKNOWN,
        observations: 'Enter here any observations, notes, that might help identify the cat more accurately.',
        mediaLinks: [],
        ...(cat ?? {}),
        volunteerName: .........
`

File: components/CatModal/CatModalView.tsx [hunk 9]
```diff
@@ -33,134 +36,162 @@ const useStyles = createStyles((theme) => ({
...
+                borderRadius: theme.radius.lg,
+                backgroundColor: !disabled ? theme.colors.green[2] : theme.colors.gray[3],
+                '&:hover': {
+                  backgroundColor: !disabled ? theme.colors.green[3] : theme.colors.gray[3],
+                },
+              }}
+              disabled={disabled}
+              onDrop={() => {}}
+              //TODO: add image upload
+              maxSize={3 * 1024 ** 2}
```
Comment (vasileorza) [inline components/CatModal/CatModalView.tsx:130]: Please move the number to a constant

**Review by vasileorza — CHANGES_REQUESTED** [review]

**Pull request merged**